### PR TITLE
Add a get_point method for Alpha_complex_3d.

### PR DESCRIPTION
**Pull request by VincentRouvreau**
- Add a get_point method for Alpha_complex_3d.
- Change Point_3 type to be either Weighted_point_3 or Bare_point_3 (new type).
- Add some get_point method tests.

### CODE REVIEW

File: src/Alpha_complex/benchmark/Alpha_complex_3d_benchmark.cpp [hunk 1]
```diff
@@ -71,10 +71,10 @@ void benchmark_points_on_torus_3D(const std::string& msg) {
   for (int nb_points = 1000; nb_points <= 125000; nb_points *= 5) {
     std::cout << "  Alpha complex 3d on torus with " << nb_points << " points." << std::endl;
     std::vector<K::Point_d> points_on_torus = Gudhi::generate_points_on_torus_3D<K>(nb_points, 1.0, 0.5);
-    std::vector<typename Alpha_complex_3d::Point_3> points;
+    std::vector<typename Alpha_complex_3d::Bare_point_3> points;
```
Comment (mglisse) [inline src/Alpha_complex/benchmark/Alpha_complex_3d_benchmark.cpp]: Why change it here? I would stick to the nicer Point_3 by default, and only use Bare/Weighted in the weighted case.

File: src/Alpha_complex/include/gudhi/Alpha_complex_3d.h [hunk 2]
```diff
@@ -567,9 +573,32 @@ Weighted_alpha_complex_3d::Weighted_point_3 wp0(Weighted_alpha_complex_3d::Point
     return true;
   }
 
+  /** \brief get_point returns the point corresponding to the vertex given as parameter.
+   *
+   * @param[in] vertex Vertex handle of the point to retrieve.
+   * @return The point found.
+   * @exception std::out_of_range In case vertex is not found (cf. std::vector::at).
+   */
+  const Point_3& get_point(std::size_t vertex) {
```
Comment (mglisse) [inline src/Alpha_complex/include/gudhi/Alpha_complex_3d.h]: Not making this method `const` (with a mutable `cgal_vertex_iterator_vector`) advertises the fact that this is not thread-safe, ok.

File: src/Alpha_complex/include/gudhi/Alpha_complex_3d.h [hunk 3]
```diff
@@ -567,9 +573,32 @@ Weighted_alpha_complex_3d::Weighted_point_3 wp0(Weighted_alpha_complex_3d::Point
...
+    if (map_cgal_simplex_tree.size() != cgal_vertex_iterator_vector.size()) {
+      cgal_vertex_iterator_vector.resize(map_cgal_simplex_tree.size());
+      for (auto map_iterator = map_cgal_simplex_tree.begin(); map_iterator != map_cgal_simplex_tree.end(); map_iterator++) {
```
Comment (mglisse) [inline src/Alpha_complex/include/gudhi/Alpha_complex_3d.h]: Still not fond of range-for loops?

File: src/Alpha_complex/include/gudhi/Alpha_complex_3d.h [hunk 4]
```diff
@@ -264,13 +269,13 @@ Weighted_alpha_complex_3d::Weighted_point_3 wp0(Weighted_alpha_complex_3d::Point
  public:
   /** \brief Alpha_complex constructor from a list of points.
    *
-   * @param[in] points Range of points to triangulate. Points must be in `Alpha_complex_3d::Point_3` or
+   * @param[in] points Range of points to triangulate. Points must be in `Alpha_complex_3d::Bare_point_3` or
    * `Alpha_complex_3d::Weighted_point_3`.
```
Comment (mglisse) [inline src/Alpha_complex/include/gudhi/Alpha_complex_3d.h]: So aren't they always `Alpha_complex_3d::Point_3`?

File: src/Alpha_complex/include/gudhi/Alpha_complex_3d.h [hunk 5]
```diff
@@ -225,30 +225,35 @@ class Alpha_complex_3d {
...
  *
  * Note: This type is defined to void if Alpha complex is not weighted.
  *
  * */
   using Weighted_point_3 = typename Triangulation_3<Kernel, Tds, Weighted, Periodic>::Weighted_point_3;
 
+  /** \brief `Alpha_complex_3d::Point_3` type is either a `Alpha_complex_3d::Bare_point_3` (Weighted = false) or a
+   * `Alpha_complex_3d::Weighted_point_3` (Weighted = true).
+   */
+  using Point_3 = typename std::conditional<Weighted, Weighted_point_3, Bare_point_3>::type;
```
Comment (mglisse) [inline src/Alpha_complex/include/gudhi/Alpha_complex_3d.h]: Is it just `typename Alpha_shape_3::Point`?

**Comment by mglisse** [comment]
> This is the reason why the vertices may be not contiguous.

I think this was copied from the dD case but doesn't apply here, because we do not use the order of the input points as vertex handle in the complex. I expect it is always contiguous with the current 3d code.

File: src/Alpha_complex/include/gudhi/Alpha_complex_3d.h (hunk 5 above)
Comment (VincentRouvreau) [reply src/Alpha_complex/include/gudhi/Alpha_complex_3d.h]: From my point of view, it is better to explain that Point_3 is a Bare_point_3 or a Weighted_point_3, than using a nested Alpha_shape type, no ? I might be wrong...

Comment (mglisse) [reply src/Alpha_complex/include/gudhi/Alpha_complex_3d.h]: We could do both: for instance keep your doc, and change the code.

**Comment by mglisse** [comment]
> I expect it is always contiguous with the current 3d code.

At some point (not this PR) we could even consider promising this, so people can use a contiguous simplex tree for it.

File: src/Alpha_complex/include/gudhi/Alpha_complex_3d.h (hunk 5 above)
Comment (VincentRouvreau) [reply src/Alpha_complex/include/gudhi/Alpha_complex_3d.h]: Done on 6d2e5d1cbdecd86ad4d56fc8902542caca2cf2bd

**Pull request merged**